Makefile

```diff
@@ -307,30 +310,39 @@ functional-with-fault-injection-test: clean-test-results
 
 test: unit-test integration-test functional-test functional-with-fault-injection-test
 
-##### Coverage #####
+##### Coverage & Reporting #####
 $(COVER_ROOT):
 	@mkdir -p $(COVER_ROOT)
 
-unit-test-coverage: $(COVER_ROOT)
+$(REPORT_ROOT):
+	@mkdir -p $(REPORT_ROOT)
+
+prepare-coverage-test: update-gotestsum $(COVER_ROOT) $(REPORT_ROOT)
+
+unit-test-coverage: prepare-coverage-test
 	@printf $(COLOR) "Run unit tests with coverage..."
-	@echo "mode: atomic" > $(UNIT_COVER_PROFILE)
-	@go test ./$(UNIT_TEST_DIRS) -timeout=$(TEST_TIMEOUT) -race $(TEST_TAG) -coverprofile=$(UNIT_COVER_PROFILE) || exit 1;
+	@gotestsum --junitfile $(NEW_REPORT) -- \
+		$(UNIT_TEST_DIRS) -timeout=$(TEST_TIMEOUT) -race $(TEST_TAG) -coverprofile=$(NEW_COVER_PROFILE) || exit 1;
```

Comment:
does anyone know what the `|| exit 1` is for? I think that can be deleted

Reply:
Yeah, I think so, too. I couldn't answer why it's there so I didn't want to remove it. But if you feel the same way, I'll axe it.